android-project/app/src/main/java/oceantreasur/es/fragments/ActivityMain.java

```diff
@@ -1,144 +1,152 @@
 
-        for(int i = 0; i < MAX_BIG_FISHES; i++) {
-            bigFishViews.add(Li.inflate(big_fish, null));
-            relativeLayouts[0].addView(bigFishViews.get(i));
-        }
-    }
+                    builder.setNeutralButton("OK", new DialogInterface.OnClickListener() {
+                        @Override
+                        public void onClick(DialogInterface dialog, int which) {
+                            MainFragment mf = new MainFragment();
+                            attachFragment(mf, "MAIN_FRAGMENT");
+                        }
+                    });
 
-    private void moveAnimation() {
-        setupAnimation(smallFishViews, smallFishBeginOffset);
-        setupAnimation(bigFishViews, bigFishBeginOffset);
-    }
+                    final AlertDialog dialog = builder.create();
+                    LayoutInflater inflater = getLayoutInflater();
+                    View dialogLayout = inflater.inflate(R.layout.alert_dialog, null);
+                    dialog.setView(dialogLayout);
+                    dialog.requestWindowFeature(Window.FEATURE_NO_TITLE);
 
-    private void setupAnimation(ArrayList<View> views, int offset) {
-        for(int i = 0; i < views.size(); i++) {
-            startAnimation(views.get(i), generateRandomIntegerInRange(MIN_DURATION, MAX_DURATION), offset);
-        }
-    }
+                    dialog.show();
 
-    private int choseRandomYPosition() {
-        int height = getScreenHeight();
+                    dialog.setOnShowListener(new DialogInterface.OnShowListener() {
+                        @Override
+                        public void onShow(DialogInterface d) {
+                            Context context = OceanTreasuresApplication.getStaticContext();
+                            ImageView image = (ImageView) dialog.findViewById(R.id.iv_dialog);
 
-        int yPosition = generateRandomIntegerInRange(1, height);
+                            Bitmap icon = BitmapFactory.decodeResource(context.getResources(),
+                                    R.drawable.fish);
 
-        if(yPosition > height * MAX_SPAWNING_POINT_IN_PERCENTS) {
-            yPosition -= (getDimensionOfFishInPx(small_fish_height) + FISH_MARGIN_BOTTOM);
-        } else {
-            if(yPosition < height * MIN_SPAWNING_POINT_IN_PERCENTS) {
-                yPosition += (getDimensionOfFishInPx(small_fish_height) - FISH_MARGIN_TOP);
+                            float imageWidthInPX = (float)image.getWidth();
+
+                            LinearLayout.LayoutParams layoutParams = new LinearLayout.LayoutParams(Math.round(imageWidthInPX),
+                                    Math.round(imageWidthInPX * (float)icon.getHeight() / (float)icon.getWidth()));
+
+                            image.setLayoutParams(layoutParams);
+                        }
+                    });
+
+                } else {
+                    GameFragment gameFragment =  (GameFragment) fragmentManagaer.findFragmentByTag("GAME_FRAGMENT");
```

Comment:
Don't use tags at all. Use  findFragmentById() or keep a reference to the active fragment in the activity.